src/main/scala-2.12/org/apache/james/gatling/imap/scenari/ImapListMessagesBodyStructureScenario.scala

```diff
@@ -2,33 +2,33 @@ package org.apache.james.gatling.imap.scenari
 
 import com.linagora.gatling.imap.PreDef._
 import com.linagora.gatling.imap.protocol.command.FetchAttributes.AttributeList
-import com.linagora.gatling.imap.protocol.command.MessageRange.From
+import com.linagora.gatling.imap.protocol.command.MessageRange.Range
 import com.linagora.gatling.imap.protocol.command.MessageRanges
 import io.gatling.core.Predef._
 import io.gatling.core.feeder.FeederBuilder
 import io.gatling.core.structure.ScenarioBuilder
 import org.apache.james.gatling.jmap.MailboxName
+import org.apache.james.gatling.jmap.scenari.ListMessageBodyStructure
 
 import scala.concurrent.duration._
 
 class ImapListMessagesBodyStructureScenario {
-  val random = scala.util.Random
 
-  val listMessagesBodyStructure = exec(imap("select").select("${mailboxName}").check(ok))
-    .exec(imap("fetch").fetch(MessageRanges(From(1)), AttributeList("BODYSTRUCTURE")).check(ok))
-
-  def generate(feeder: FeederBuilder, mailboxes: Iterable[MailboxName], mailboxesListed: Int): ScenarioBuilder =
+  def generate(feeder: FeederBuilder, mailboxes: Iterable[MailboxName], mailsFetched: Int): ScenarioBuilder =
     scenario("ImapListMessagesBodyStructureScenario")
       .feed(feeder)
       .pause(1.second)
       .exec(imap("Connect").connect()).exitHereIfFailed
       .exec(imap("login").login("${username}", "${password}").check(ok))
-      .exec{session =>
-        session.set("mailboxes", random.shuffle(mailboxes.map(_.name)).take(mailboxesListed))
-      }
-      .group("Fetch messages BODYSTRUCTURE for mailboxes") {
-        exec(foreach("${mailboxes}", "mailboxName") {
-          exec(listMessagesBodyStructure)
-        })
+      .exec(_.set("mailboxes", mailboxes.map(_.name)))
+      .group(ListMessageBodyStructure.name) {
+        exec(
+          exec(imap("select")
+            .select("${mailboxes.random()}")
+            .check(ok))
+          .exec(imap("fetch")
+            .fetch(MessageRanges(Range(1, mailsFetched)), AttributeList("BODYSTRUCTURE"))
```

Comment:
I'd rather take some non-contiguous ranges but if you find it to be overkill, I can live with that version